data_model/src/query.rs

```diff
@@ -103,6 +105,8 @@ pub enum QueryBox {
     FindPermissionTokensByAccountId(FindPermissionTokensByAccountId),
     /// [`FindAllPermissionTokenDefinitions`] variant.
     FindAllPermissionTokenDefinitions(FindAllPermissionTokenDefinitions),
+    /// [`DoesAccountHavePermissionToken`] variant.
+    DoesAccountHavePermissionToken(DoesAccountHavePermissionToken),
```

Comment:
This query might make sense even though we have `FindPermissionTokensByAccountId`

Reply:
I'd err on the side of not adding extra specialised code unless we really know what we're optimising.

Reply:
This should work faster than `FindPermissionTokensbyAccountId` because we don't need them all. We just need to check one invariant.

So I don't get it, do you think it's okay or do you think I should remove it?

Reply:
I would remove it but it depends on how many tokens will be returned by `FindPermissionTokensbyAccountId`? Seems like premature optimization

Reply:
> So I don't get it, do you think it's okay or do you think I should remove it?

I would keep it. 

It's a fast lookup and you're not optimising anything prematurely, you're just adding a thing that's fast in one specific way.

Reply:
Alright, let's keep it then